examples/bicluster/plot_bicluster_newsgroups.py

```diff
@@ -41,18 +41,21 @@
 print(__doc__)
 
 
-def number_aware_tokenizer(doc):
-    """ Tokenizer that maps all numeric tokens to a placeholder.
+def number_normalizer(tokens):
+    """ Map all numeric tokens to a placeholder.
 
     For many applications, tokens that begin with a number are not directly
     useful, but the fact that such a token exists can be relevant.  By applying
     this form of dimensionality reduction, some methods may perform better.
     """
-    token_pattern = re.compile(u'(?u)\\b\\w\\w+\\b')
-    tokens = token_pattern.findall(doc)
-    tokens = ["#NUMBER" if token[0] in "0123456789_" else token
-              for token in tokens]
-    return tokens
+    for t in tokens:
+        yield re.sub(r"^\d+$", "#NUMBER", t)
```

Comment:
On this dataset there seem to be quite a few tokens that start with digits and should probably be discarded, so it might be a good idea to showcase this idea, as well as to keep behavior as close to the old example as possible.

(I'm ok with removing the underscore, that was not really explained in the old example)